pkg/domain/resourcegroup/runaway.go

```diff
@@ -480,38 +513,47 @@ func newRunawayChecker(
 }
 
 // BeforeExecutor checks whether query is in watch list before executing and after compiling.
-func (r *RunawayChecker) BeforeExecutor() error {
+func (r *RunawayChecker) BeforeExecutor() (string, error) {
```

Comment:
How about `BeforeExecutor(stmt Ctx)`?

Reply:
It can be, but introducing `*stmtctx.StatementContext` in `RunawayChecker` will have the circular dependency problem. Though we could use an interface like `resourceGroupNameSetter` to resolve this, it's also simple to just directly return the switch resource group here.